platform-framework/analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala

```diff
@@ -45,12 +45,16 @@ object CommonUtil {
 
     def getSparkContext(parallelization: Int, appName: String, sparkCassandraConnectionHost: Option[AnyRef] = None, sparkElasticsearchConnectionHost: Option[AnyRef] = None): SparkContext = {
         JobLogger.log("Initializing Spark Context")
-        val conf = new SparkConf().setAppName(appName);
+        val conf = new SparkConf().setAppName(appName)
         val master = conf.getOption("spark.master");
         // $COVERAGE-OFF$ Disabling scoverage as the below code cannot be covered as they depend on environment variables
         if (master.isEmpty) {
             JobLogger.log("Master not found. Setting it to local[*]")
             conf.setMaster("local[*]");
+            conf.set("spark.default.parallelism", parallelization.toString)
```

Comment:
These configurations should be set outside of the master configuration check. This should not be combined with the master.isEmpty check as we can run the spark jobs in distributed mode later.